--- Torr_Samaho-zandronum-00065f166c52.orig/src/gl/r_render/r_render.h	2012-08-19 22:11:18.000000000 +0200
+++ Torr_Samaho-zandronum-00065f166c52/src/gl/r_render/r_render.h	2012-12-22 11:53:40.208356293 +0100
@@ -1,6 +1,13 @@
 #ifndef R_RENDER
 #define R_RENDER
 
+#ifndef PFNGLMULTITEXCOORD2FPROC
+	typedef void (APIENTRYP PFNGLMULTITEXCOORD2FPROC) (GLenum target, GLfloat s, GLfloat t);
+#endif
+#ifndef PFNGLMULTITEXCOORD2FVPROC
+	typedef void (APIENTRYP PFNGLMULTITEXCOORD2FVPROC) (GLenum target, const GLfloat *v);
+#endif
+
 enum RenderFlags
 {
 	RFL_NPOT_TEXTURE=1,
