https://bugs.launchpad.net/screenruler/+bug/1900364

--- a/ruler_window.glade
+++ b/ruler_window.glade
@@ -11,7 +11,6 @@
     <property name="default_height">36</property>
     <property name="type_hint">dialog</property>
     <property name="decorated">False</property>
-    <property name="has_resize_grip">False</property>
     <signal name="button-press-event" handler="on_button_press_event" swapped="no"/>
     <signal name="key-press-event" handler="on_key_press_event" swapped="no"/>
     <signal name="delete-event" handler="on_delete_event" swapped="no"/>
--- a/ruler_window.rb
+++ b/ruler_window.rb
@@ -243,7 +243,7 @@
 
 		# Loop, drawing ticks (top and bottom) and labels
 		repetitions, tick_index = 0, 0
-		loop(pixels_per_tick, length + OVERDRAW, pixels_per_tick) { |x|
+		loopn(pixels_per_tick, length + OVERDRAW, pixels_per_tick) { |x|
 			x = x.floor + 0.5		# Cairo likes lines in the 'center' of pixels
 
 			tick_size = @@tick_sizes[ unit.tick_pattern[tick_index, 1].to_s ]
@@ -332,7 +332,7 @@
 
 		# Fill with 'horizontal' lines
 		cr.set_source_color($preferences_window.foreground_color)
-		loop(@menu_box.y + 2.5, @menu_box.y + @menu_box.height + -1.5, 2) { |y|
+		loopn(@menu_box.y + 2.5, @menu_box.y + @menu_box.height + -1.5, 2) { |y|
 			cr.move_to(@menu_box.x + 2.0, y)
 			cr.line_to(@menu_box.x + @menu_box.width - 1, y)
 		}
--- a/utils/addons_ruby.rb
+++ b/utils/addons_ruby.rb
@@ -59,7 +59,7 @@
 		list.each { |file| orig_require(file) }
 	end
 
-	def loop(from, to, step=1)
+	def loopn(from, to, step=1)
 		i = from
 		while i <= to
 			yield i
