From 21857c4115bee6bbebabac2ffde647ef450b7013 Mon Sep 17 00:00:00 2001
From: trinity <imandry.c@gmail.com>
Date: Sat, 21 Dec 2013 17:21:35 +0800
Subject: [PATCH] Accept MIME application/x-wacom-tablet

---
 src/common.h | 3 +++
 src/main.c   | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/common.h b/src/common.h
index c4d5283..f079216 100644
--- a/src/common.h
+++ b/src/common.h
@@ -47,6 +47,9 @@
 #define PLUGIN_VERSION "2.0.0.1" // Highest version of API that is supported.
 #define PLUGIN_VERSIONL 2001     // Decimal version of above.
 #define PLUGIN_MIME    "application/x-wacomtabletplugin"
+#define PLUGIN_MIME2   "application/x-wacom-tablet"
+#define PLUGIN_MIME_DESCRIPTION \
+    PLUGIN_MIME "::" PLUGIN_NAME ";" PLUGIN_MIME2 "::" PLUGIN_NAME
 
 // Just a helper macro, makes for nicer looking code.
 #define IS_IDENTIFIER(id) (name == g_netscapeFuncs->getstringidentifier(id))
diff --git a/src/main.c b/src/main.c
index 01d9fc2..91c966a 100644
--- a/src/main.c
+++ b/src/main.c
@@ -46,7 +46,7 @@ const char* NP_GetPluginVersion(void)
 // Plugin MIME + description.
 const char* NP_GetMIMEDescription(void)
 {
-  return PLUGIN_MIME"::"PLUGIN_NAME;
+  return PLUGIN_MIME_DESCRIPTION;
 }
 
 // Called when plugin is loaded.
