--- src/baobab-window.c.orig	2015-04-12 18:19:09.357337252 +0200
+++ src/baobab-window.c	2015-04-12 18:19:33.237338193 +0200
@@ -724,7 +724,7 @@ BaobabWindow* baobab_window_construct (G
 	gtk_container_add ((GtkContainer*) self, G_TYPE_CHECK_INSTANCE_TYPE (_tmp58_, GTK_TYPE_WIDGET) ? ((GtkWidget*) _tmp58_) : NULL);
 	_tmp59_ = _ ("Disk Usage Analyzer");
 	gtk_window_set_title ((GtkWindow*) self, _tmp59_);
-	gtk_window_set_hide_titlebar_when_maximized ((GtkWindow*) self, TRUE);
+	gtk_window_set_hide_titlebar_when_maximized ((GtkWindow*) self, FALSE);
 	_baobab_location_unref0 (self->priv->active_location);
 	self->priv->active_location = NULL;
 	self->priv->scan_completed_handler = (gulong) 0;
